commands/chassis/ChassisShift.java

```diff
@@ -10,44 +10,25 @@
  */
 public class ChassisShift extends Command {
 	protected final SolenoidShifters solenoids;
-	protected SolenoidShifters.ShiftState state;
+	protected final SolenoidShifters.ShiftState state;
 	
 	/**
-	 * Shifts the solenoids to the opposite state
+	 * Shifts the solenoids to the state state
 	 *
 	 * @param solenoids
+	 * @param state
 	 */
-	public ChassisShift(SolenoidShifters solenoids) {
+	public ChassisShift(SolenoidShifters solenoids, SolenoidShifters.ShiftState state) {
```

Comment:
Actually, anything that called the constructor `public ChassisShift(SolenoidShifters solenoids) {` would set shift to null, and cause a nullpointerexception on the first switch for it